gamestonk_terminal/portfolio_optimization/port_opt_api.py

```diff
@@ -0,0 +1,360 @@
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+        weights = {}
+        values = {}
+        n_stocks = len(list_of_stocks)
+        for stock in list_of_stocks:
+            weights[stock] = round(1 / n_stocks, 5)
+            values[stock] = ns_parser.value * round(1 / n_stocks, 5)
+        if ns_parser.pie:
+            pie_chart_weights(values)
+        return values
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def property_weighting(
+    list_of_stocks: List[str], property_type: str, other_args: List[str]
+):
+    """
+    Property weighted portfolio where each weight is the relative fraction.
+    Parameters
+    ----------
+    list_of_stocks: List[str]
+        List of tickers to be included in optimization
+    property_type: str
+        Property to weight by.  Can be anything in yfinance.Ticker().info.  Examples:
+            "marketCap", "dividendYield", etc
+
+    Returns
+    -------
+    weights: dict
+        Dictionary of weights where keys are the tickers
+    """
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="market_cap_weighted",
+        description="Return portfolio weights/values that are weighted by marketcap",
+    )
+
+    parser.add_argument(
+        "-v",
+        "--value",
+        default=1,
+        type=float,
+        dest="value",
+        help="Amount to allocate to portfolio",
+    )
+    parser.add_argument(
+        "--pie",
+        action="store_true",
+        dest="pie",
+        default=False,
+        help="Display a pie chart for weights",
+    )
+    weights = {}
+    prop = {}
+    prop_sum = 0
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+        for stock in list_of_stocks:
+            stock_prop = yf.Ticker(stock).info[property_type]
+            if stock_prop is None:
+                stock_prop = 0
+            prop[stock] = stock_prop
+            prop_sum += stock_prop
+        for k, v in prop.items():
+            weights[k] = round(v / prop_sum, 5) * ns_parser.value
+        if ns_parser.pie:
+            pie_chart_weights(weights)
+        return weights
+
+    except Exception as e:
+        print(e)
+        print("")
+        return
+
+
+def show_ef(list_of_stocks: List[str], other_args: List[str]):
+    parser = argparse.ArgumentParser(add_help=False, prog="ef")
+
+    parser.add_argument(
+        "-p",
+        "--period",
+        default="3mo",
+        dest="period",
+        help="period to get yfinance data from",
+        choices=period_choices,
+    )
+    parser.add_argument(
+        "-n", default=300, dest="n_port", help="number of portfolios to simulate"
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return {}
+        stock_prices = process_stocks(list_of_stocks, ns_parser.period)
```

Comment:
I think to follow MVC pattern you need to have a function this `argparse` and then call a function, that is the one that generates the plot

Reply:
not following....

Reply:
Don't worry with this one, it doesn't matter!